### PR TITLE
ROX-17163: bump GHA CI images to a version with Go 1.19.9 for 3.74.x

### PR DESCRIPTION
**Pull request by tommartensen**
## Description

* Bump Go version from 1.18.4 to 1.19.9
  * Update expected Go version
  * Update GHA & test images (rox-ci-image) to 0.3.57.1

## Checklist
- [ ] Investigated and inspected CI test results -> CI 
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Starting (& failing) CI is sufficient. Failed tests will be fixed in follow-up PRs. 

Known issues:
* style-checks will complain about code comment formatting. To resolve, changes from https://github.com/stackrox/stackrox/pull/4648 shall be cherry-picked.

### CODE REVIEW

**Comment by janisz** [comment]
Typo in the description. It's not 1.19.4 but 1.18.4
```
docker run quay.io/stackrox-io/apollo-ci:stackrox-test-0.3.55 go version
go version go1.18.4 linux/amd64
```

File: pkg/sac/allow_fixed_scopes_checker_core.go [hunk 1]
```diff
@@ -12,8 +12,10 @@ type allowFixedScopesCheckerCore []scopeKeySet
 // AllowFixedScopes returns a scope checker core that allows those scopes that
 // are in the cross product of all individual scope key lists. I.e.,
 // AllowFixedScopes(
-//   AccessModeScopeKeys(storage.Access_READ, storage.Access_READ_WRITE),
-//   ResourceScopeKeys(resources.CLUSTER),
+//
+//	AccessModeScopeKeys(storage.Access_READ, storage.Access_READ_WRITE),
+//	ResourceScopeKeys(resources.CLUSTER),
+//
```
Comment (janisz) [inline pkg/sac/allow_fixed_scopes_checker_core.go:18]: This has wrong indentation compare with master https://github.com/stackrox/stackrox/blob/master/pkg/sac/allow_fixed_scopes_checker_core.go#L15-L18

**Review by janisz — DISMISSED** [review]
Looks like not all comments are formatted correctly. Could you cherry-pick changes from https://github.com/stackrox/stackrox/pull/4648 and best split this PR into series of smaller PRs so reviewing will be easier?

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -21,7 +21,7 @@ jobs:
       ROX_PRODUCT_BRANDING: ${{ matrix.branding }}
     runs-on: ubuntu-latest
     container:
-      image: quay.io/stackrox-io/apollo-ci:stackrox-test-0.3.55
+      image: quay.io/stackrox-io/apollo-ci:stackrox-test-0.3.57.1
```
Comment (janisz) [inline .github/workflows/build.yaml:24]: Why version has `.1` suffix? Why not usual x.y.z?

**Comment by tommartensen** [comment]
@janisz Because we are under time pressure to deliver the patch release, we will ignore your request for changes to the Go comments. Our priority is to make the CI pass. 
After merging this PR, we will merge #6100 to the release branch to make the style checks pass. 

If you believe that there is value in cherry-picking #4648 to the release branch or repeating the work on a fix branch aiming to the release branch, we'd like to ask that you do this work. 

Thanks for understanding, if you have any questions about that, we'd be happy to set up a quick meeting to resolve.

**Comment by janisz** [comment]
@tommartensen cherrpick applied without any issues although I remember there were some additional style changes needed. 

-  https://github.com/stackrox/stackrox/pull/6111

File: .github/workflows/build.yaml (hunk 2 above)
Comment (tommartensen) [reply .github/workflows/build.yaml:24]: We branched out from 0.3.57 to create an image with 1.19.9, >0.3.58 are on Go 1.20 already.

**Review by gavin-stackrox — APPROVED** [review]
LGTM

**Pull request merged**